metals/src/main/scala/scala/meta/internal/metals/CodeLensProvider.scala

```diff
@@ -1,50 +1,114 @@
 package scala.meta.internal.metals
-import java.util
+
 import java.util.Collections._
+
 import ch.epfl.scala.{bsp4j => b}
 import org.eclipse.{lsp4j => l}
+
+import scala.concurrent.ExecutionContext
+import scala.meta.internal.metals.ClientCommands.StartDebugSession
+import scala.meta.internal.metals.CodeLensProvider._
 import scala.meta.internal.metals.MetalsEnrichments._
 import scala.meta.internal.mtags.Semanticdbs
 import scala.meta.io.AbsolutePath
 
 final class CodeLensProvider(
-    classes: BuildTargetClasses,
+    buildTargetClasses: BuildTargetClasses,
     buffers: Buffers,
     buildTargets: BuildTargets,
+    compilations: Compilations,
     semanticdbs: Semanticdbs
-) {
-  def findLenses(path: AbsolutePath): util.List[l.CodeLens] = {
-    buildTargets.inverseSources(path) match {
-      case Some(buildTarget) if classes.main(buildTarget).nonEmpty =>
-        findLenses(path, buildTarget).asJava
-      case _ =>
-        emptyList[l.CodeLens]()
-    }
+)(implicit ec: ExecutionContext) {
```

Comment:
Can we use explicit `ExecutionContext` in parameters?

Reply:
To explain, I think it would be useful to give explicit context to Providers so that we know what each of them are using. Will be easier to figure out if we need to optimize.

Reply:
The convention in the codebase so far has been to pass around the `ExecutionContext` implicitly, is there a reason we should do it differently for `CodeLensProvider`?

Reply:
I'm fine with implicit, I just thought it might help to have explicit ones for the providers. But should not be important currently.

Reply:
It's not possible to explicitly pass the `ExecutionContext` in for comprehensions
```scala
for {
  a <- foo()
} yield a + 2
```

Reply:
I just really meant this for constructors like here.